### PR TITLE
Feature/UC-152 uve + iteration. Iteration 4

### PR DESCRIPTION
**Pull request by Carlos-Augusto**
Defines new endpoint for the rule engine integration.

### CODE REVIEW

File: uve-api.yaml [hunk 1]
```diff
@@ -76,13 +86,74 @@ paths:
...
+            type: string
+          in: header
+          name: X-Forwarded-Tls-Client-Cert
+          description: X509 pem string. It has to be a chained certificate as comma-separated string.
+      requestBody:
+        description: The json payload representation of a certificate.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CertificatePayload'
```
Comment (Carlos-Augusto) [inline uve-api.yaml]: @mwicke We would like to agree on the payload type. We can receive the DCC payload, which is what the "verify" endpoint returns or we can receive the DCC QR Code data?

What are your thought on this

File: uve-api.yaml [hunk 2]
```diff
@@ -116,11 +187,45 @@ paths:
...
+            "is": "Ministry of Health, Austria",
+            "ci": "URN:UVCI:01:AT:10807843F94AEE0EE5093FBC254BD813#B"
+          }
+        ]
+      }
+    CertificatePayloadValidationResponse:
+      type: string
+      format: json
+      example: {
+        "DE": true,
```
Comment (Carlos-Augusto) [inline uve-api.yaml]: @mwicke this is the express result per country, of course depending on the input list of countries

File: uve-api.yaml [hunk 3]
```diff
@@ -76,13 +86,74 @@ paths:
...
+          name: X-Transaction-Tag
+          description: An optional random tag used for accounting specified by the user.
+          schema:
+            type: string
+          example: HAM-Termninal-1
+          required: false
+        - in: header
+          name: X-Validate-For
+          description: |
+            A comma-separated value of the countries to check. If no values or header is provided, All supported countries will be checked.
```
Comment (Carlos-Augusto) [inline uve-api.yaml]: @dagdelenmustafa look at this change, plz. When no value all countries are searched

File: uve-api.yaml (hunk 2 above)
Comment (Carlos-Augusto) [reply uve-api.yaml]: As per our conversation, this has been adjusted.

**Comment by Carlos-Augusto** [comment]
We have added verfiy v2 definition verifies a digital health certificate for authenticity and validates a digital health certificate against a list of rules which vary per country.

File: uve-api.yaml [hunk 4]
```diff
@@ -116,11 +264,57 @@ paths:
...
+      type: string
+      format: json
+      example: {
+        "DE": true,
+        "FR": false
+      }
+    CertificatePayloadValidationResponseComposed:
+      type: string
+      format: json
+      ##TODO: Check how to ref other schema
```
Comment (Carlos-Augusto) [inline uve-api.yaml]: We need to find a way to use a similar definition

**Comment by thinkberg** [comment]
Please document the error messages somewhere (if they are not from the rules) and lets do a cross check they are understandable.

File: uve-api.yaml (hunk 2 above)
Comment (mwicke) [reply uve-api.yaml]: Thanks, very good!

File: uve-api.yaml (hunk 1 above)
Comment (mwicke) [reply uve-api.yaml]: I would suggest to use the DCC JSON payload.

**Comment by Carlos-Augusto** [comment]
@mwicke we are considering removing the endpoint /api/uve/v1/countries/validate. From our point of view, it seems like it is not really adding value to the API at this moment. But we could be missing something? What is your take on this?

**Comment by mwicke** [comment]
@Carlos-Augusto as discussed, we can remove v1/countries/validate in the future, since the v2/verify endpoint makes this one obsolete

**Review by mwicke — APPROVED** [review]

**Pull request merged**